react/AccordionItem/AccordionItem.js

```diff
@@ -0,0 +1,105 @@
+import React, { Fragment, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import { ChevronIcon } from 'seek-style-guide/react';
+import classnames from 'classnames';
+import styles from './AccordionItem.less';
+import { toggleContent } from './utils';
+import { CLOSED_HEIGHT } from './constants';
+
+function AccordionItem({
+  className,
+  title,
+  children,
+  open,
+  onOpen,
+  onClose,
+  ...restProps
+}) {
+  const initialHeight = open ? 'auto' : CLOSED_HEIGHT;
+  const initialVisibility = open ? 'visible' : 'hidden';
+  const initialOverflow = initialVisibility;
+  const contentEl = useRef(null);
+  const [currentHeight, setCurrentHeight] = useState(initialHeight);
+  const [cssVisibility, setCssVisibility] = useState(initialVisibility);
+  const [cssOverflow, setCssOverflow] = useState(initialOverflow);
+  const [timeoutHandle, setTimeoutHandle] = useState(null);
+  const [isOpen, setIsOpen] = useState(open);
+
+  const buttonClasses = classnames(className, styles.title);
+
+  const isAnimating = currentHeight === CLOSED_HEIGHT && isOpen;
+  const expanderClasses = classnames(styles.expander, {
+    [styles.expanderOpen]: isAnimating
+  });
+
+  const chevronClasses = classnames(styles.chevron, {
+    [styles.chevronOpen]: isOpen
+  });
+
+  return (
+    <Fragment>
+      <button
+        type="button"
+        className={buttonClasses}
+        onClick={() => {
+          toggleContent({
+            el: contentEl.current,
+            setCurrentHeight,
+            timeoutHandle,
+            setTimeoutHandle,
+            isOpen,
+            setIsOpen,
+            setCssVisibility,
+            setCssOverflow
+          });
+
+          if (isOpen && onClose) {
+            onClose();
+          } else if (!isOpen && onOpen) {
+            onOpen();
+          }
+        }}
+        {...restProps}
+      >
+        {typeof title === 'string' ? (
+          <span className={styles.titleText}>{title}</span>
+        ) : (
+          <span className="AccordionItem__titleContent">{title}</span>
+        )}
+        <ChevronIcon
+          direction="down"
```

Comment:
Would it be possible to use the direction prop here to control the rotation?

Reply:
@jahredhope (see above animated gif for reference)

Reply:
Nvm, @adenj has added a commit, that works well.